quinn/src/connection.rs

```diff
@@ -622,6 +654,10 @@ where
             ref_count: 0,
         })))
     }
+
+    fn index(&self) -> usize {
+        self.0.lock().unwrap().index()
+    }
```

Comment:
Given that this requires taking a lock, perhaps we should consider just returning the address contained in the `Arc`. Can you think of any reason why we specifically need the quinn-proto level ID here?

Reply:
Yeah, I don't think it needs to specifically be the quinn-proto ID, just something that's static. `&*arc as *const _ as usize` would good enough for me.

Reply:
Let's favor the simpler/faster approach, then.

Reply:
Although `index` is a bit silly if it's not even remotely sequential/dense... maybe rename to `stable_id`?

Reply:
Yeah, I like this idea